src/data/data1d/Spectrum1D/peaks/optimizePeaks.ts

```diff
@@ -0,0 +1,29 @@
+import { PeakXYWidth } from 'cheminfo-types';
+import { xFindClosestIndex } from 'ml-spectra-processing';
+import { xyPeaksOptimization } from 'nmr-processing';
+
+import { Datum1D, Peak } from '../../../types/data1d';
+
+import { mapPeaks } from './mapPeaks';
+
+interface OptimizePeaksOptions {
+  from: number;
+  to: number;
+}
+
+export function optimizePeaks(datum1D: Datum1D, options: OptimizePeaksOptions) {
+  const { from, to } = options;
+  let { re, x } = datum1D.data;
+
+  if (from !== undefined && to !== undefined) {
+    const fromIndex = xFindClosestIndex(datum1D.data.x, from);
+    const ToIndex = xFindClosestIndex(datum1D.data.x, to);
+
+    x = x.slice(fromIndex, ToIndex);
```

Comment:
Why making a copy rather than subarray as suggested in the issue ?

Reply:
this is my mistake